odd suggestion for comma typo

In an obscure situation with a typo (where there is a comma instead of a period), rustc will suggest replacing the comma with a semicolon, which will fail to compile.

```rust
let a = std::process::Command::new("echo")
    .arg("1")
    ,arg("2")
    .output();
```

Produces the following error and suggestion:

```
error: expected `;`, found ``,``
 --> src/main.rs:4:5
  |
4 |     ,arg("2")
  |     ^ help: change this to `;`: `;`

error[E0425]: cannot find function `arg` in this scope
 --> src/main.rs:4:6
  |
4 |     ,arg("2")
  |      ^^^ not found in this scope

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
```

Some observations:

* This must be a let binding.
* The method calls must be on separate lines (all on one line gives a different error).

I would expect this to provide the same error if it is all on one line ("expected one of `.`, `;`, `?`, or an operator"), possibly with a "maybe incorrect" suggestion to replace it with a period (maybe only if the following character is not whitespace?).

Note: Feel free to close this.  This is really obscure, and I suspect it is not trivial to change, and it is fairly obvious what is wrong.  I just thought it was strange, and I was doubly puzzled that the indentation mattered.

### Meta
rustc 1.45.0-nightly (9912925c2 2020-05-10)


<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"chrissimpkins"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/src/librustc_parse/parser/diagnostics.rs
+++ b/src/librustc_parse/parser/diagnostics.rs
@@ -934,6 +934,19 @@ impl<'a> Parser<'a> {
             return self.expect(&token::Semi).map(drop);
         } else if !sm.is_multiline(self.prev_token.span.until(self.token.span)) {
             // The current token is in the same line as the prior token, not recoverable.
+        } else if [token::Comma, token::Colon].contains(&self.token.kind)
+            && &self.prev_token.kind == &token::CloseDelim(token::Paren)
+        {
+            // Likely typo: The current token is on a new line and is expected to be
+            // `.`, `;`, `?`, or an operator after a close delimiter token.
+            //
+            // let a = std::process::Command::new("echo")
+            //         .arg("1")
+            //         ,arg("2")
+            //         ^
+            // https://github.com/rust-lang/rust/issues/72253
+            self.expect(&token::Semi)?;
+            return Ok(());
         } else if self.look_ahead(1, |t| {
             t == &token::CloseDelim(token::Brace) || t.can_begin_expr() && t.kind != token::Colon
         }) && [token::Comma, token::Colon].contains(&self.token.kind)

--- a/src/test/ui/issues/issue-72253.rs
+++ b/src/test/ui/issues/issue-72253.rs
@@ -1,0 +1,6 @@
+fn main() {
+    let a = std::process::Command::new("echo")
+        .arg("1")
+        ,arg("2") //~ ERROR expected one of `.`, `;`, `?`, or an operator, found `,`
+        .output();
+}

--- a/src/test/ui/issues/issue-72253.stderr
+++ b/src/test/ui/issues/issue-72253.stderr
@@ -1,0 +1,10 @@
+error: expected one of `.`, `;`, `?`, or an operator, found `,`
+  --> $DIR/issue-72253.rs:4:9
+   |
+LL |         .arg("1")
+   |                  - expected one of `.`, `;`, `?`, or an operator
+LL |         ,arg("2")
+   |         ^ unexpected token
+
+error: aborting due to previous error
+

